### PR TITLE
adds a check to see if storage container is secured/locked on disposal units

### LINKED ISSUE #15270: Emptying a locked secure briefcase into a disposal bin makes all the contents inside pour out.
As the title says, all you need to do to reproduce this is have a locked secure briefcase (mine had 10k credits in 1000 credit notes), click on a disposal bin, and all the contents inside end up inside the bin, while the briefcase remains locked.

### PR DESCRIPTION
**Pull request by harryob**
closes #15270

### CODE REVIEW

**Comment by harryob** [comment]
!review

**Comment by harryob** [comment]
!bugfix

**Comment by alsoandanswer** [comment]
you should use Fixes #15270 , it automatically makes it so the issue becomes closed once this PR gets merged.

**Comment by harryob** [comment]
> you should use "Fixes #15270" , it automatically makes it so the issue becomes closed once this PR gets merged.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
not to worry, closes does the same

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -177,6 +177,13 @@
 
 	if(istype(I, /obj/item/storage) && length(I.contents) && user.a_intent != I_HURT)
 		var/obj/item/storage/S = I
+
+		if(istype(S, /obj/item/storage/secure))
+			var/obj/item/storage/secure/secured_storage = S
+			if(secured_storage.locked)
+				to_chat(user, SPAN_WARNING("You can't empty \the [secured_storage] into \the [src] - it is locked."))
```
Comment (SleepyGemmy) [inline code/modules/recycling/disposal.dm:184]:
```suggestion
				to_chat(user, SPAN_WARNING("You can't empty \the [secured_storage] into \the [src]. It is locked."))
```

File: html/changelogs/harryob-disposalbriefcase.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+author: harryob
+
+delete-after: True
+
+changes:
+  - bugfix: "disposal units no longer bypass high tech locking devices."
```
Comment (SleepyGemmy) [inline html/changelogs/harryob-disposalbriefcase.yml:6]:
```suggestion
  - bugfix: "Disposal units no longer bypass high tech locking devices."
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**